src/controllers/bulk/bulkcontroller.cpp

```diff
@@ -175,31 +171,18 @@ int BulkController::open() {
         qCWarning(m_logBase) << "Unable to open USB Bulk device" << getName();
         return -1;
     }
-
-#if defined(__WINDOWS__) || defined(__APPLE__)
-    if (m_interfaceNumber && libusb_kernel_driver_active(m_phandle, m_interfaceNumber) == 1) {
-        qCDebug(m_logBase) << "Found a driver active for" << getName();
-        if (libusb_detach_kernel_driver(m_phandle, 0) == 0)
-            qCDebug(m_logBase) << "Kernel driver detached for" << getName();
-        else {
-            qCWarning(m_logBase) << "Couldn't detach kernel driver for" << getName();
-            libusb_close(m_phandle);
-            return -1;
-        }
+    if (libusb_set_auto_detach_kernel_driver(m_phandle, true) == LIBUSB_ERROR_NOT_SUPPORTED) {
+        qCDebug(m_logBase) << "unable to automatically detach kernel driver for" << getName();
     }
 
-    if (m_interfaceNumber) {
-        int ret = libusb_claim_interface(m_phandle, m_interfaceNumber);
-        if (ret < 0) {
-            qCWarning(m_logBase) << "Cannot claim interface for" << getName()
-                                 << ":" << libusb_error_name(ret);
-            libusb_close(m_phandle);
-            return -1;
-        } else {
-            qCDebug(m_logBase) << "Claimed interface for" << getName();
-        }
+    if (int ret = libusb_claim_interface(m_phandle, m_interfaceNumber); ret < 0) {
```

Comment:
Let's avoid this confusing style with an assignment in an if condition.

Reply:
Why is this confusing? Its just a new syntax which makes sense here since I don't need `ret` outside the if-body
https://en.cppreference.com/w/cpp/language/if#if_statements_with_initializer

Reply:
Because it break the flow when reading the code. You expect an if condition like everywhere else but you read the assignment, which is every where else in the line before. This reduces IMHO maintainability and lead to misunderstandings. The benefit of the tight scope is negligible here.

Reply:
> The benefit of the tight scope is negligible here.

I disagree actually because if the variable does not get its own scope the variables storing the return values either need different names or the one variable needs to be reassigned. The different names is confusing because that leads me to believe that we're actually interested in this value while reassigning looks weird because the added mutability makes the code hard to reason about. 

> You expect an if condition like everywhere else but you read the assignment, which is every where else in the line before.

Well you only expect the condition there because you're not used to there being something else but this is a well designed and documented feature so its reasonable to expect it to be usable. The same applies to other "new" features such as lambdas or trailing return types.

Let me reiterate that this very much distinct from the relying on return value of `operator=` (eg. `if (int var = f()`) because that is mixing an init-statement with a condition while the latter one is clearly defined.

This is just a case of "Was der Bauer nicht kennt, frisst er nicht". 

I won't insist on using this style, but resisting to embrace new and encouraged features and styles is a common pattern in code review. This is neither rewarding nor productive.

Reply:
I have considered this a bit and it turns out that this is a general code style question, we should discuss on Zulip. 

We have a plenty of places where we may consider to tighten the scope of a local variable, because we can. On the other hand we don't care about them, because it will just introduce boiler plate and nesting, interrupting the flow when reading the code. This is an issue when you only skim over the code, and especially when the if condition is broken over two lines of code.

I have no interest to ban this new statement from the Mixxx code entirely, but let's use it deliberately, backed up by a rule in our style guide. 

This is here an edge case because we have two return values that will end up in the same variable of temporary nature otherwise. For now this is a new style, that does not fit to the rest of Mixxx. I am open to decide in any direction, but  let's do it consistently. 

I will not block the PR on this point, but prefer reusing `ret` here without  a scope.

Reply:
https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/C.2B.2B17.20if.20with.20init-statement